pkg/detectors/aws/aws.go

```diff
@@ -101,6 +103,34 @@ func GetHMAC(key []byte, data []byte) []byte {
 	return hasher.Sum(nil)
 }
 
+func GetAccountNumFromAWSID(AWSID string) (string, error) {
+	// Function to get the account number from an AWS ID (no verification required)
+	// Source: https://medium.com/@TalBeerySec/a-short-note-on-aws-key-id-f88cc4317489
+	if len(AWSID) < 4 {
+		return "", fmt.Errorf("AWSID is too short")
+	}
+	trimmed_AWSID := AWSID[4:]
+	x, err := base32.StdEncoding.WithPadding(base32.NoPadding).DecodeString(strings.ToUpper(trimmed_AWSID))
+	if err != nil {
+		return "", err
+	}
+
+	if len(x) < 6 {
+		return "", fmt.Errorf("Decoded AWSID is too short")
+	}
+	y := x[0:6]
```

Comment:
suggestion: You can make a few optimizations here. We can pre-alloc a buffer to reduce allocations and handle BE format.
ex:
```
data := make([]byte, 8)
copy(data[2:], decodedBytes[0:6]) // Only the first 6 bytes are relevant.
```